lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -72,13 +73,12 @@ class MockTimers {
   }
 
   async * #setIntervalPromisified(interval, startTime, ...args) {
-    while (true) {
-      yield startTime;
-      await new Promise((resolve) => {
-        this.#createTimer(false, resolve, interval, ...args);
-      });
+    const intervalEmitter = new EventEmitter();
+    const callback = () => intervalEmitter.emit('data', startTime);
+    this.#createTimer(true, callback, interval, ...args);
 
-      startTime += interval;
+    for await (const events of EventEmitter.on(intervalEmitter, 'data')) {
```

Comment:
does EventEmitter.on's return value have a nonconfigurable nonwritable Symbol.asyncIterator?

Reply:
It doesn't, why?

Reply:
then for-await-of loops aren’t safe (in a primordials sense)

Reply:
> then for-await-of loops aren’t safe (in a primordials sense)

got it! Gonna have to make my own extend then like:

```js
const EventEmitter = require('events');

class AsyncEventEmitter extends EventEmitter {
  [Symbol.asyncIterator]() {
    const events = [];
    const iterator = {
      next: async () => {
        if (events.length > 0) {
          return { done: false, value: events.shift() };
        }

        return new Promise((resolve) => {
          const eventHandler = (event) => {
            resolve({ done: false, value: event });
            this.off('event', eventHandler);
          };

          this.on('event', eventHandler);
        });
      },
      return: () => {
        this.removeAllListeners();
        return { done: true };
      },
    };

    return iterator;
  }
}

const emitter = new AsyncEventEmitter();
emitter.on('event', () => console.log('Event emitted'));

(async () => {
  for await (const event of emitter) {
    console.log(event);
  }
})();
```

Reply:
As long as no users can ever get to AsyncEventEmitter, or, if you `Object.defineProperty` it to be nonconfigurable and nonwritable, then that should work fine.

Reply:
perfect! Should I add it to primordials so we can use it later?

Reply:
done! I did it a bit differently would you check it?